Node start

andreimatei opened the pull request.

andreimatei committed:
storage: plumb a ctx to sotrate.NewStore()

Release note: None

andreimatei committed:
server: remove an initStores() method

That method was quite pitiful, a shell of some 2014 glory. Its comment
was wrong, one argument was unused, one return value was always nil.
The code is better without its 3 lines too.

Release note: None

andreimatei committed:
server: rename a param

Release note: None

andreimatei committed:
sever,storage: minor refactorings around bootstrapping

Release note: None

andreimatei committed:
server: remove unused code from bootstrapCluster

There was some really nasty code that's luckily now unused (I think
likely it became unused in #33150).
More rationalizing of that method coming next.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33630)
<!-- Reviewable:end -->

andreimatei committed:
storage: get rid of VersionRangeAppliedStateKey

This migration is complicated. The patch gets rid of the version guard,
but not otherwise the migration code. Before this patch, the migration
for a range was triggered when a range was not migrated yet and
IsMinSupported(VersionRangeAppliedStateKey). The latter part is always
true: since a previous commit set BinaryMinimumSupportedVersion=2_1,
VersionRangeAppliedStateKey was guaranteed.
After this patch, the migration is triggered when a range was not yet
migrated.
It would seem that we have to live with this migration for an undefined
period of time longer: some ranges might have never been migrated if
they originated in an, say, 1.1 cluster that went through a series of
upgrades but the range was never touched.

One thing that was considered was having the below-Raft command
application always perform the migration if the range wasn't previously
migrated, regardless of what the proposal specifies. This doesn't seem
feasible since we need to guarantee that different replicas apply the
command exactly the same, and so we can't have one ignore the proposal
spec and another one not ignore it. We'll be able to unconditionally
perform the migration once we decree that there can no longer be
proposals "in flight" coming from ancient nodes (think unapplied Raft
lags).

Release note: None

andreimatei committed:
cluster: bump BinaryMinimumSupportedVersion=Version2_1 from 2_0

... making 2.2 nodes incompatible with 2.0 nodes.
This lets me assume 2.1 nodes (and do the next commit). The instructions
for when to bump this say to bump this when we introduce 2_2, but I
think that's off (or, rather, we should already have introduced 2.2).
Also see #33578.

Release note: None

andreimatei committed:
storage: move the WriteInitialData() method out of Store

That method is used for writing initial data to an engine. Before this
patch it was using a Store, but for no good reason. This meant that we
had to create a Store only to stop it immediately and re-create it later
with the same engine.
This patch makes the method free-standing and it now takes an engine.

Release note: None

andreimatei committed:
pkg: remove an unused argument in bootstrapCluster()

Release note: None

andreimatei committed:
wip on moving init out of server

Release note: None

andreimatei committed:
server: rework the initServer

Before this patch, the initServer was complicated - it had some sort of
a semaphore exposed with an unclear API. It was also performing
bootstrapping of the cluster, duplicating code with the Server which
performs bootstrap in cases where an initServer is not used.
No more. This patch reduces the scope of the initServer to be a
glorified channel signaled when bootstrap is requested. Bootstrapping is
left to the calling Server, and so code is unified.

This patch is part of a somewhat chaotic assault on the cluster
bootstrap and generally node startup code, which I find inscrutable.
Other patches flying by stem from this unravelling sweater. One of the
goals is to rework bootstrapping such that the nodeIDContainer hack goes
away.

Release note: None

andreimatei committed:
gossip: add convenience methods for dealing with gossipping the cluster id

The Gossip already is handles the cluster id key specially, so I've
added more convenience methods for dealing with it. Besides convenience,
they also make it clear that this info doesn't have a TTL.

Release note: None

bobvawter commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=33630) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=33630) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=33630) it.</sub>
